tests/fixtures/core.py

```diff
@@ -472,6 +472,30 @@ def project(test_dir, project_init_service):
         logging.debug(f"Cleaned project at {project.root}")
 
 
+# Copied the fixture above in order to change the scope
+# Frees us from side affects from other tests in the same class
+@pytest.fixture(scope="function")
```

Comment:
Instead of copying the body of the function, could you please move the body into a new [`contextmanager`](https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager) function, then have the fixtures each use it like so:

```python
@pytest.fixture(scope="my scope")
def fixture_name(...):
    with my_context_manager():
        yield
```

Reply:
Yes, much better! Thank you